saleor/graphql/product/tests/test_product.py

```diff
@@ -301,6 +310,64 @@ def test_fetch_unavailable_products(user_api_client, product):
     assert not content["data"]["products"]["edges"]
 
 
+def test_fetch_all_products_visible_in_listings(
+    user_api_client, product_list, permission_manage_products,
+):
+    # given
+    product_list[0].visible_in_listings = False
+    product_list[0].save(update_fields=["visible_in_listings"])
+
+    product_count = Product.objects.count()
+
+    # when
+    response = user_api_client.post_graphql(QUERY_FETCH_ALL_PRODUCTS)
+
+    # then
+    content = get_graphql_content(response)
+    product_data = content["data"]["products"]["edges"]
+    assert len(product_data) == product_count - 1
```

Comment:
Don't you think that we should compare here at least the returned IDs? Or confirm that we didn't get product with `visible_in_listings == False`
Something like :
`assert id_of_product_not_visible not in product_data[IDs]` ?